### ISSUE TITLE
Duplicated database definitions

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#689]
There are two sources for creating the MySQL (Maria DB) or PostgreSQL database, respectively:

```
share/initial-mysql.sql
share/initial-postgres.sql
```
```
script/create_db_mysql.pl
script/create_db_postgresql_9.3.pl
```

The installation instructions refer to the `.sql` files, whereas Travis scripts use the Perl scripts. Do we need both or could we decide on one variant? With two there is a high risk of deviation.

We already have a deviation for MySQL (Maria DB):

In `script/create_db_mysql.pl`: `results blob DEFAULT NULL,`

In `share/initial-mysql.sql`: `results mediumblob DEFAULT NULL,`

The latter was introduced to resolve an issue with large results.

**Comment by mattias-p** [comment]
I agree that two definitions for one database is one definition too much.
Keeping the fixed version of the MySQL definition seems like the reasonable option.

**Comment by matsduf** [comment]
@pnax, can you take a look at this?

**Comment by pnax** [comment]
> @pnax, can you take a look at this?

yes, totally. Any preference on keeping the commands in a Perl script (or in a subroutine as for [SQLite](https://github.com/zonemaster/zonemaster-backend/blob/6f32539ab07639be9d552b35c551cf4941e740ee/lib/Zonemaster/Backend/DB/SQLite.pm#L52)) or within an `.sql` file?

**Comment by matsduf** [comment]
One aspect is ease of writing, and then an `.sql` file is probably the easiest. For the user a perl script makes it easier, but an `.sql` file can be input to a perl script. I do not have a strong opinion, but it should be clear, and having all three database engines the same way is a plus.

**Comment by pnax** [comment]
Any idea what is the benefit to use a `SEQUENCE` here instead of a `SERIAL PRIMARY KEY`?

https://github.com/zonemaster/zonemaster-backend/blob/6f32539ab07639be9d552b35c551cf4941e740ee/script/create_db_postgresql_9.3.pl#L43 

https://github.com/zonemaster/zonemaster-backend/blob/6f32539ab07639be9d552b35c551cf4941e740ee/share/initial-postgres.sql#L8

If they are both equivalent, I'll stick with the definition in `initial-postgresql.sql` due to the fact that this is the file used in [installation document](https://github.com/zonemaster/zonemaster-backend/blob/master/docs/Installation.md#322-instructions-for-postgresql-centos).

**Comment by matsduf** [comment]
> Any idea what is the benefit to use a `SEQUENCE` here instead of a `SERIAL PRIMARY KEY`?

@mtoma, do you know?

**Comment by pnax** [comment]
I started working on this and I realized that we need to create the MySQL/PostgreSQL database at some point. I haven't find an easy way to do it with the DBI (as we need extra privileges).

The pro to use a script is that we can reuse the values from the config (for the database, user and password).
The con is that I cannot find an easy way to create the database (if you have any idea or suggestion on how this can be done, I'd be happy to hear them).

So I think I'll remove the scripts for now and keep the `.sql` files. And see if I can add a wrapper to properly set the database name, the user and password by getting them from the config.
